src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/RequestPayloadValueResolver.php

```
@@ -0,0 +1,132 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\Controller\ArgumentResolver;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Attribute\MapQueryString;
+use Symfony\Component\HttpKernel\Attribute\MapRequestBody;
+use Symfony\Component\HttpKernel\Controller\ValueResolverInterface;
+use Symfony\Component\HttpKernel\ControllerMetadata\ArgumentMetadata;
+use Symfony\Component\HttpKernel\Exception\HttpException;
+use Symfony\Component\Serializer\Exception\NotEncodableValueException;
+use Symfony\Component\Serializer\Exception\UnsupportedFormatException;
+use Symfony\Component\Serializer\Normalizer\AbstractObjectNormalizer;
+use Symfony\Component\Serializer\Normalizer\DenormalizerInterface;
+use Symfony\Component\Serializer\SerializerInterface;
+use Symfony\Component\Validator\Exception\ValidationFailedException;
+use Symfony\Component\Validator\Validator\ValidatorInterface;
+
+/**
+ * @author Konstantin Myakshin <molodchick@gmail.com>
+ */
+final class RequestPayloadValueResolver implements ValueResolverInterface
+{
+    private const DEFAULT_FORMAT = 'json';
+
+    private const CONTEXT_DENORMALIZE = [
+        AbstractObjectNormalizer::DISABLE_TYPE_ENFORCEMENT => true,
+        DenormalizerInterface::COLLECT_DENORMALIZATION_ERRORS => true,
+    ];
+
+    private const CONTEXT_DESERIALIZE = [
+        DenormalizerInterface::COLLECT_DENORMALIZATION_ERRORS => true,
+    ];
+
+    public function __construct(
+        private readonly ?SerializerInterface $serializer,
+        private readonly ?DenormalizerInterface $denormalizer,
```

see my comment above: I don't think we should make those two nullable - they're not optional deps

---

to improve DX - if `symfony/serializer` and Request Attribute is used - we are not silently ignoring it but provide suggestion to install missing component